### PR TITLE
[SPARK-15529] [SQL] Replace SQLContext and HiveContext with SparkSession in Test

### PR DESCRIPTION
**Pull request by gatorsmile**
#### What changes were proposed in this pull request?

This PR is to use the new entrance `Sparksession` to replace the existing `SQLContext` and `HiveContext` in SQL test suites.

No change is made in the following suites:
- `ListTablesSuite` is to test the APIs of `SQLContext`. 
- `SQLContextSuite` is to test `SQLContext`
- `HiveContextCompatibilitySuite` is to test `HiveContext`

**Update**: Move tests in `ListTableSuite` to `SQLContextSuite`
#### How was this patch tested?

N/A

### CODE REVIEW

**Comment by rxin** [comment]
What's the reason that ListTablesSuite should not be ported?

**Comment by gatorsmile** [comment]
Originally, `ListTableSuite` was added to test the API `tables` in `SQLContext`
```Scala
  def tables(databaseName: String): DataFrame = {
    Dataset.ofRows(sparkSession, ShowTablesCommand(Some(databaseName), None))
  }
```

In `SparkSession`, we do not have such an API. Instead, I think the new way for retrieving a list of tables is [`spark.catalog.listTables`](https://github.com/apache/spark/blob/0f61d6efb45b9ee94fa663f67c4489fbdae2eded/sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala#L85-L107). This API has already been tested in the suite [`CatalogSuite.scala` ](https://github.com/apache/spark/blob/845e447fa03bf0a53ed79fa7e240af94dc152d2c/sql/core/src/test/scala/org/apache/spark/sql/internal/CatalogSuite.scala#L133-L174) 

Basically, the only value of the test suite `ListTableSuite` is to verify `tables` in `SQLContext`. Feel free to let me know if you want to remove it? keep it? or modify it using the new API? Thanks!

**Comment by rxin** [comment]
Maybe move it into SQLContextSuite ?

**Comment by gatorsmile** [comment]
Sure, will do. Thanks!

**Comment by rxin** [comment]
Merging in master/2.0. Thanks.

**Comment by rxin** [comment]
Can we also update the test suites used in machine learning?

**Comment by gatorsmile** [comment]
Sure, will do. Thanks!
